### PR TITLE
extend-analyzehandler

### PR DESCRIPTION
**Pull request by xueningl-az**
Add new interface for Solution Analyze.

### CODE REVIEW

File: src/PortingAssistant.Client.Client/Reports/ReportExporter.cs [hunk 1]
```diff
@@ -82,6 +82,8 @@ public ReportExporter(ILogger<ReportExporter> logger)
 
                     //project apis analsis result
                     string ApiAnalyzeFileName = ProjectName + "-api-analysis.json";
+                    projectAnalysResult.SourceFileAnalysisResults.Sort(
+                        (r1, r2) => string.Compare(r1.SourceFileName, r2.SourceFileName, StringComparison.OrdinalIgnoreCase));
```
Comment (jonlouie) [inline src/PortingAssistant.Client.Client/Reports/ReportExporter.cs:86]: There was a performance regression identified- do we need to sort here?

File: src/PortingAssistant.Client.Client/Reports/ReportExporter.cs [hunk 2]
```diff
@@ -66,9 +67,8 @@ public ReportExporter(ILogger<ReportExporter> logger)
                     Directory.CreateDirectory(FileDir);
                     List<PackageAnalysisResult> packageAnalysisResults = new List<PackageAnalysisResult>();
                     Dictionary<PackageVersionPair, string> packageAnalysisResultErrors = new Dictionary<PackageVersionPair, string>();
-
-                    projectAnalysResult.PackageAnalysisResults.ToList()
-                    .ForEach(p =>
+                    var sorted = projectAnalysResult.PackageAnalysisResults.OrderBy(c=>c.Key.PackageId).ToList();
```
Comment (jonlouie) [inline src/PortingAssistant.Client.Client/Reports/ReportExporter.cs:70]: There was a performance regression identified- do we need to sort here?

Comment (xueningl-az) [reply src/PortingAssistant.Client.Client/Reports/ReportExporter.cs:70]: i need to compare the result for packages in order.  sorted will make it easier.  but can revert it back

File: src/PortingAssistant.Client.Client/Reports/ReportExporter.cs (hunk 1 above)
Comment (xueningl-az) [reply src/PortingAssistant.Client.Client/Reports/ReportExporter.cs:86]: make compare between liteb and b more easy

**Review by jonlouie — APPROVED** [review]

**Review by huawenm — APPROVED** [review]

**Pull request merged**